src/deploy/functions/ensure.ts

```diff
@@ -0,0 +1,131 @@
+import * as clc from "cli-color";
+
+import { ensure } from "../../ensureApiEnabled";
+import { FirebaseError, isBillingError } from "../../error";
+import { logLabeledBullet, logLabeledSuccess } from "../../utils";
+import { ensureServiceAgentRole } from "../../gcp/secretManager";
+import { defaultServiceAccount } from "../../gcp/cloudfunctions";
```

Comment:
You're going to need a v1 and v2 here. The default service account for v1 functions is the GAE service account and for v2 functions is the compute service account.

Reply:
This make sense, but since overall v2 support is going to be nontrivial addition to the code, I'm thinking of tackling in a separate PR (will make a separate cf3-secrets branch). WDTY?

Reply:
Went ahead and implemented the fn to support both v1 and v2 platform.